### PR TITLE
Added clarification on timing side channel

### PR DESCRIPTION
**Pull request by meriac**
@AlessandroA @Patater - see https://github.com/ARMmbed/uvisor-helloworld/issues/48

### CODE REVIEW

File: source/box-challenge.cpp [hunk 1]
```diff
@@ -62,7 +62,17 @@ static bool secure_compare(const uint8_t *src, const uint8_t *dst, int len)
 
 UVISOR_EXTERN bool __verify_secret(const uint8_t *secret, int len)
 {
-    /* only accept the right challenge size */
+    /* We only accept calls where len according to our expectations.
```
Comment (AlessandroA) [inline source/box-challenge.cpp]: where len *meets* our expectations.

**Comment by Patater** [comment]
LGTM

Prefer "Add" instead of "Added" in the commit message.

File: source/box-challenge.cpp [hunk 2]
```diff
@@ -62,7 +62,16 @@ static bool secure_compare(const uint8_t *src, const uint8_t *dst, int len)
 
 UVISOR_EXTERN bool __verify_secret(const uint8_t *secret, int len)
 {
-    /* only accept the right challenge size */
+    /* We only accept calls where 'len@ equals CHALLENGE_SIZE.
```
Comment (Patater) [inline source/box-challenge.cpp]: Actually, replace @ with '.

**Comment by meriac** [comment]
@AlessandroA, @Patater - updated according to your recommendations

**Comment by Patater** [comment]
LGTM

**Pull request merged**